packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,242 @@
+        stdin.setRawMode(true);
+        readline.emitKeypressEvents(stdin);
+
+        // Configure keypress event for actions
+        const actions = Object.keys(this.keys);
+        stdin.on('keypress', (_, actionKey) => {
+            const possibleActions = actions.filter((action) => {
+                return this.keys[action] === actionKey.name;
+            });
+
+            if (possibleActions.length === 0) {
+                return;
+            }
+
+            if (possibleActions.length > 1) {
+                throw new Error('Multiple actions are provided for same key');
+            }
+
+            const action = possibleActions[0];
+            this.handlers[action](compiler);
+        });
+
+        if (compiler.compilers) {
+            this.isMultiCompiler = true;
+            this.compilers = compiler.compilers;
+        }
+
+        // Register Custom Hook for printing after clrscr
+        if (!compiler.hooks.beforeInteractiveStats) {
+            compiler.hooks = {
+                ...compiler.hooks,
+                beforeInteractiveStats: new SyncHook(),
+            };
+        }
+
+        // Clear for first run as well
+        clrscr();
+        compiler.hooks.beforeInteractiveStats.call();
+
+        if (!this.isMultiCompiler) {
+            // Clear output on watch invalidate
+            compiler.hooks.beforeCompile.tap(this.name, () => {
+                clrscr();
+                compiler.hooks.beforeInteractiveStats.call();
+            });
+
+            compiler.hooks.afterDone.tap(this.name, () => {
+                setTimeout(() => {
+                    console.log('\n\n');
+                    spawnCommand('compilation completed', true);
+                }, 1);
+            });
+        } else {
+            const helperPlugin = new InteractiveModeMultiCompilerHelperPlugin();
+
+            // Register helper plugin on each of child compiler
+            for (const childCompiler of this.compilers) {
+                helperPlugin.apply(childCompiler);
+            }
+
+            compiler.hooks.done.tap(this.name, () => {
+                const allDone = this.compilers.reduce((result, childCompiler) => {
+                    return result && !childCompiler.watching.running;
+                }, true);
+
+                if (!allDone) return;
+
+                setTimeout(() => {
+                    console.log('\n\n');
+                    spawnCommand('all compilations completed', true);
+                }, 100);
+            });
+        }
+    }
+
+    quitHandler(compiler) {
+        if (this.isMultiCompiler) {
+            for (const childCompiler of this.compilers) {
+                if (childCompiler.watching === undefined) continue;
+                childCompiler.watching.close();
+            }
+            process.exit(0);
+        }
+
+        if (compiler.watching === undefined) return;
+        compiler.watching.close(() => {
+            process.exit(0);
+        });
+        return;
+    }
```

Comment:
No need to extra logic here always use `compiler.close()`, we need run it for single build for cache purpose too

Reply:
Also do not forget handle errors here